src/Shared/HttpSys/RequestProcessing/NativeRequestContext.cs

```diff
@@ -650,14 +650,14 @@ private void GetUnknownHeadersHelper(IDictionary<string, StringValues> unknownHe
 
     private SocketAddress? GetEndPointHelper(bool localEndpoint, HTTP_REQUEST_V1* request, byte* pMemoryBlob)
     {
-        var source = localEndpoint ? (byte*)request->Address.pLocalAddress : (byte*)request->Address.pRemoteAddress;
+        var source = localEndpoint ? request->Address.pLocalAddress : request->Address.pRemoteAddress;
 
         if (source == null)
         {
             return null;
         }
-        var address = (IntPtr)(pMemoryBlob + _bufferAlignment - (byte*)_originalBufferAddress + source);
```

Comment:
Is it really ok to delete this address fixup? It seems to be required given the buffer gymnastics employed by this type.

(The other places in this file use `long fixup` argument instead of `byte* pMemoryBlob` argument that is a bit easier to understand in my opinion.)

Reply:
Good advice. Even if the line can be deleted (which I believed from tracing its call sites and seeing that all the args canceled each other out), it's probably not in scope of this PR to start refactoring the code base outside of SocketAddress.